### PR TITLE
User model/routes/controller/views

### PR DESCRIPTION
**Pull request by mstoople532**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new
+    @user.email = user_params[:email]
```
Comment (rposborne) [inline app/controllers/users_controller.rb:8]: Do we need to be this explicit for these attributes?  Would it be better if we pass `user_params` directly into `User.new`

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+  validates :password, confirmation: true
+  validates :password_confirmation, presence: true
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  validates :email, presence: true, format: { with: VALID_EMAIL_REGEX }
+
+  def password
```
Comment (rposborne) [inline app/models/user.rb]: Should we use `has_secure_password`?

File: config/routes.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :users, only: [:new, :create]
```
Comment (rposborne) [inline config/routes.rb]: Resources implies that we will manage `:id` in our routes, but new/create do not need id's?  Could we get away with just `resource`?

File: test/fixtures/users.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+one:
```
Comment (rposborne) [inline test/fixtures/users.yml]: Could we add a fixture for each of the roles that we have discussed? `staff`, `client`, `admin`

File: test/controllers/users_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UsersControllerTest < ActionDispatch::IntegrationTest
```
Comment (rposborne) [inline test/controllers/users_controller_test.rb:3]: Could you add basic tests for the controller actions that you added?

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (mstoople532) [reply app/controllers/users_controller.rb:8]: I don't think we actually lose anything by leaving this explicit - it's easier to read and know at a glance exactly what parameters are being created

File: app/models/user.rb (hunk 2 above)
Comment (mstoople532) [reply app/models/user.rb]: Yes.

**Review by Rdore88 — APPROVED** [review]
Looks good

**Comment by mstoople532** [comment]
@Rdore88 i fixed the merge conflicts with schema, can you take another look and approve?

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (rposborne) [reply app/controllers/users_controller.rb:8]: Does strong parameters not cover this explicit nature, if we have to write out the valid attributes these feels like we filter once and then filter the same thing again.

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  has_secure_password
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  validates :email, presence: true, format: { with: VALID_EMAIL_REGEX }
+
+  def password
+    @password ||= BCrypt::Password.new(password_digest)
+  end
+
+  def password=(plaintext_password)
```
Comment (rposborne) [inline app/models/user.rb]: Does this method conflict with `has_secure_password`?  I think it defines a method `User#password=`

**Review by rposborne — APPROVED** [review]

**Pull request merged**